src/map/water_supply.c

```diff
@@ -108,9 +107,8 @@ static void fill_aqueducts_from_offset(int grid_offset)
         map_aqueduct_set_water_access(grid_offset, 1);
         int image_id = map_image_at(grid_offset);
         if (map_terrain_is(grid_offset, TERRAIN_HIGHWAY)) {
-            if (!had_water_access) {
-                map_image_set(grid_offset, image_id - 18);
-            }
+            image_id = city_draw_highway_get_aqueduct_image(grid_offset);
```

Comment:
I think that instead of setting the tile image to the aqueduct, you can remove this code and tie the tile image to the highway image instead.

In fact this code here is redundant because you're checking for the proper aqueduct image in the draw code, when drawing over the highway.

Reply:
Ok so it’s probably not at all obvious from the code but the sneaky thing I’m doing for aqueducts is I’m still setting the image ids on highways so the tops draw correctly. The new highway rendering code is only applied for footprints.

Reply:
That is a good idea, but still, I don't think you need to set the tile image to the aqueduct top beforehand for that.

Reply:
If you don't set it it looks like this as soon as the water supply updates:
![image](https://user-images.githubusercontent.com/2081059/200189114-6a65bdc5-f2db-43dc-b620-11a4ce7e5eb9.png)

Reply:
Good enough for me! Looks like present-day ruins hehe